packages/compass-logging/src/logger.ts

```diff
@@ -47,22 +49,26 @@ export function createGenericLoggerAndTelemetry(
     event: string,
     properties: TrackProps = {}
   ): Promise<void> => {
-    // Avoid circular dependency between compass-logging and compass-preferences-model
-    // Note that this is mainly a performance optimization, since the main process
-    // telemetry code also checks this preference value, so it is safe to fall back to 'true'.
-    try {
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore-error Types from the dependency may not be available in early bootstrap.
-      preferences ??= (await import('compass-preferences-model'))
-        .preferencesAccess;
-    } catch {
-      preferences ??= {
-        getPreferences() {
-          return { trackUsageStatistics: true };
-        },
-      };
+    if (!preferencesService) {
```

Comment:
> Maybe we can drop this check and always use a safe fallback instead of this?

Soo … I definitely agree that this should be the state after COMPASS-7407, but until we get to that, there are still a lot of places where something just calls `createLoggerAndTelemetry(component)`, and that is the case which we’re accounting for here. As the comment says, it’s only a performance optimization, so yeah, we could get rid of this entire bit (and also the corresponding parts in `provider`) if we want to.

Reply:
Ahhh, gotcha, this PR is only the "initial" part, understood! I would maybe prefer to have a TODO for when we can remove it or would drop it already if you are definitely okay with that. Don't have the strongest preference, but this code is a bit confusing to figure out and so would be nice to remove as soon as we have a chance I think 🙂

Reply:
Sure, added a TODO comment :+1: